### PR TITLE
Cloaked mobs don't give off visible messages (+cloak logic refactor)

### PR DESCRIPTION
**Pull request by Zonespace27**
# About the pull request
Pretty much the title. Visible messages (visible_message proc) don't give off messages to others if the origin is cloaked.

I'm a bit leery of doing this on `visible_message` so this is gonna need testing first.

Makes checking for cloaking better, adding a trait on cloak instead of checking for user alpha

# Explain why it's good for the game
Being able to see things like "Unknown peers through the laser designator" while they're invisible is a pretty large tell despite you not having seen them.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Works

</details>

# Changelog
:cl:
balance: Cloaked mobs no longer give off visible messages
/:cl:

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -186,6 +190,8 @@
 	var/view_dist = 7
 	if(max_distance) view_dist = max_distance
 	for(var/mob/M as anything in viewers(view_dist, src))
+		if(HAS_TRAIT(src, TRAIT_CLOAKED))
+			continue
 		M.show_message(message, SHOW_MESSAGE_VISIBLE, blind_message, SHOW_MESSAGE_AUDIBLE, message_flags)
```
Comment (realforest2001) [inline code/modules/mob/mob.dm:195]:
```suggestion
	if(HAS_TRAIT(src, TRAIT_CLOAKED))
		return FALSE
	var/view_dist = 7
	if(max_distance)
		view_dist = max_distance
	for(var/mob/M as anything in viewers(view_dist, src))
		M.show_message(message, SHOW_MESSAGE_VISIBLE, blind_message, SHOW_MESSAGE_AUDIBLE, message_flags)
	return TRUE
```

**Review by realforest2001 — CHANGES_REQUESTED** [review]

**Comment by realforest2001** [comment]
Can you add the cloak trait to the changelog/title as it's the main part of this PR

**Comment by Zonespace27** [comment]
> Can you add the cloak trait to the changelog/title as it's the main part of this PR

Added to title; changelog isn't needed due to it not being player-facing

**Review by realforest2001 — APPROVED** [review]
lgtm

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 2]
```diff
@@ -120,6 +120,7 @@
 	if (LPA && istype(LPA))
 		LPA.knockdown = TRUE // pounce knocks down
 		LPA.freeze_self = TRUE
+	ADD_TRAIT(bound_xeno, TRAIT_CLOAKED, "lurker_cloak")
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:123]: trait source defines

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 3]
```diff
@@ -130,6 +131,7 @@
 		LPA.knockdown = FALSE // pounce no longer knocks down
 		LPA.freeze_self = FALSE
 	bound_xeno.stealth = FALSE
+	REMOVE_TRAIT(bound_xeno, TRAIT_CLOAKED, "lurker_cloak")
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:134]: trait source defines

File: code/modules/cm_preds/yaut_bracers.dm [hunk 4]
```diff
@@ -571,6 +571,7 @@
 			return FALSE
 
 		cloaked = TRUE
+		ADD_TRAIT(M, TRAIT_CLOAKED, "pred_cloak")
```
Comment (harryob) [inline code/modules/cm_preds/yaut_bracers.dm:574]: trait source defines

File: code/modules/cm_preds/yaut_bracers.dm [hunk 5]
```diff
@@ -617,7 +618,8 @@
 		cloak_malfunction = world.time + decloak_timer
 
 	cloaked = FALSE
-	log_game("[key_name_admin(usr)] has disabled their cloaking device.")
+	REMOVE_TRAIT(user, TRAIT_CLOAKED, "pred_cloak")
```
Comment (harryob) [inline code/modules/cm_preds/yaut_bracers.dm:621]: trait source defines

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**